### PR TITLE
Purchases Sample: Login

### PR DESCRIPTION
**Pull request by beylmk**
Adds login/reset functionality.

Also:
* cleans up PurchaserInfo card to display less
* adds button to manage subscriptions
* ability to copy user ID (for logging back in)
* add labels to current offering and active subscription cards

The test build is now active in the Play Store. Let me know if you want me to add you to the testers list.

### CODE REVIEW

File: purchases-sample/src/main/java/com/revenuecat/sample/OfferingCardAdapter.kt [hunk 1]
```diff
@@ -19,14 +23,16 @@ class OfferingCardAdapter(private val offerings: List<Offering>, private val lis
...
+        holder.bind(offering, offering == currentOffering)
     }
 
     inner class OfferingViewHolder(private val binding: OfferingCardBinding) :
         RecyclerView.ViewHolder(binding.root) {
 
-        fun bind(offering: Offering) {
+        fun bind(offering: Offering, isCurrent: Boolean) {
             binding.offering = offering
+            binding.isCurrent = isCurrent
```
Comment (vegaro) [inline purchases-sample/src/main/java/com/revenuecat/sample/OfferingCardAdapter.kt:35]: Where's this variable defined? I can't find it in the xml

File: purchases-sample/src/main/java/com/revenuecat/sample/LoginFragment.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    lateinit var binding: FragmentLoginBinding
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = FragmentLoginBinding.inflate(inflater)
+
+        binding.loginButton.setOnClickListener {
+            binding.loginUsernameEditText.text?.toString()?.let { userId ->
+                Purchases
+                    .sharedInstance
```
Comment (vegaro) [inline purchases-sample/src/main/java/com/revenuecat/sample/LoginFragment.kt:25]: I would have the `.sharedInstance` in the previous line

File: purchases-sample/src/main/java/com/revenuecat/sample/OfferingFragment.kt [hunk 3]
```diff
@@ -48,16 +60,16 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
...
-                    Toast.LENGTH_LONG
-                ).show()
+            Toast.makeText(
+                requireContext(),
+                "Successful purchase, order id: ${purchaseDetails.orderId}",
+                Toast.LENGTH_LONG
+            ).show()
 
-                findNavController().navigateUp()
+            findNavController().navigateUp()
```
Comment (vegaro) [inline purchases-sample/src/main/java/com/revenuecat/sample/OfferingFragment.kt:72]: I think I would have the onSuccess lambda as another parameter so this indentation doesn't look weird:

```
      Purchases.sharedInstance.purchasePackageWith(
            requireActivity(),
            currentPackage,
            { error, userCancelled ->
                if (!userCancelled) {
                    showUserError(requireActivity(), error)
                }
            },
            { purchaseDetails, _ ->
                Toast.makeText(
                    requireContext(),
                    "Successful purchase, order id: ${purchaseDetails.orderId}",
                    Toast.LENGTH_LONG
                ).show()

                findNavController().navigateUp()
            })
            ```

File: purchases-sample/src/main/res/layout/fragment_overview.xml [hunk 4]
```diff
@@ -9,6 +9,8 @@
         <variable
                 name="purchaserInfo"
                 type="com.revenuecat.purchases.PurchaserInfo" />
+
+        <import type="android.view.View" />
```
Comment (vegaro) [inline purchases-sample/src/main/res/layout/fragment_overview.xml:13]: what's this for?

File: purchases-sample/src/main/java/com/revenuecat/sample/OfferingCardAdapter.kt (hunk 1 above)
Comment (beylmk) [reply purchases-sample/src/main/java/com/revenuecat/sample/OfferingCardAdapter.kt:35]: in offering_card.xml!

File: purchases-sample/src/main/res/layout/fragment_overview.xml (hunk 4 above)
Comment (beylmk) [reply purchases-sample/src/main/res/layout/fragment_overview.xml:13]: that's so i can access View.VISIBLE when setting visibility on the manage button

**Comment by aboedo** [comment]
We may wanna start calling it purchaseTester: 
- to better represent its purpose
- so that customers don't try to use it as a sample given that they should be using MagicWeather for that

We did the rename a couple of weeks ago on iOS for the same reason

**Comment by aboedo** [comment]
not needed for this PR, though, we can do it as a separate thing

**Comment by beylmk** [comment]
> We may wanna start calling it purchaseTester:
> 
> * to better represent its purpose
> * so that customers don't try to use it as a sample given that they should be using MagicWeather for that
> 
> We did the rename a couple of weeks ago on iOS for the same reasons

I can include that in the final PR into develop!

File: purchases-sample/src/main/res/layout/fragment_overview.xml (hunk 4 above)
Comment (vegaro) [reply purchases-sample/src/main/res/layout/fragment_overview.xml:13]: Oh I see

**Review by vegaro — APPROVED** [review]
Looks good to me!

**Pull request merged**